lib/ddtrace/contrib/action_cable/events/transmit.rb

```diff
@@ -32,7 +32,6 @@ def span_type
           def process(span, _event, _id, payload)
             channel_class = payload[:channel_class]
 
-            span.service = configuration[:service_name]
             span.resource = channel_class
             span.span_type = span_type
```

Comment:
I'm curious -- there seems to be quite a lot of boilerplate on these events; e.g. the logic in `process` changes a bit, but almost everything else is repeated boilerplate.

In particular, setting the `span.span_type` from the `#span_type` seems like something we always want, and we could do a similar trick for the component -- would it be possible to move this to the caller or some common place so that it would be one thing less to repeat?

Reply:
I agree that these are super repetitive.
I've never got the time to actually refactor ActiveSupport event handling integrations though.

Reply:
@ivoanjo what do you think of? (Different event, but same idea)
```ruby
module Datadog
  module Contrib
    module ActionCable
      module Events
        class PerformAction < ActionCable::RootContextEvent.new(
            event_name: 'perform_action.action_cable',
            span_name: Ext::SPAN_ACTION,
            span_type: Datadog::Ext::AppTypes::WEB
          )

          def self.process(span, _event, _id, payload)
            channel_class = payload[:channel_class]
            action = payload[:action]

            span.resource = "#{channel_class}##{action}"

            Contrib::Analytics.set_measured(span)

            span.set_tag(Ext::TAG_CHANNEL_CLASS, channel_class)
            span.set_tag(Ext::TAG_ACTION, action)
          end
        end
      end
    end
  end
end
```

Reply:
Yeah, I had something like that in mind.

Some further suggestions:
1. If only the content of `process` changes, perhaps we could even just pass it as a block to `new`.
2. Do we need to make it a class or module, or could it just be a constant? E.g. `PERFORM_ACTION = ...Event.new(...)`?

Reply:
Actually, yesterday I was watching a Rails 7 presentation and activerecord hass a style we could use as well; their example was:

```ruby
class CreatePosts < ActiveRecord::Migration[7.0]
  def change
    # ...
  end
end
```